File: app/src/main/java/de/fhkiel/temi/robogguide/database/Locations.java
package de.fhkiel.temi.robogguide.database;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

@DatabaseTable(tableName = "locations")
public class Locations {

    @DatabaseField(generatedId = true)
    private int id;

    @DatabaseField
    private String name;

    @DatabaseField
    private boolean important;

    @DatabaseField
    private int places_id;

    public Locations() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean isImportant() {
        return important;
    }

    public void setImportant(boolean important) {
        this.important = important;
    }

    public int getPlaces_id() {
        return places_id;
    }

    public void setPlaces_id(int places_id) {
        this.places_id = places_id;
    }
}

File: app/src/main/java/de/fhkiel/temi/robogguide/database/Locations.kt
package de.fhkiel.temi.robogguide.database

import com.j256.ormlite.field.DatabaseField
import com.j256.ormlite.table.DatabaseTable

/**
 * Orm Data class to get the locations table
 *
 * @property id
 * @property name
 * @property important
 * @property places_id
 */
@DatabaseTable(tableName = "locations")
data class Locations(

    @DatabaseField(generatedId = true, columnName = "id")
    val id: Int,

    @DatabaseField(columnName = "name")
    var name: String,

    @DatabaseField(columnName = "important")
    val important: Int,

    @DatabaseField(columnName = "places_id")
    val places_id: Int

) {
    constructor() : this(0,"",0,0)
}